src/source.F90

```diff
@@ -162,8 +156,21 @@ subroutine sample_external_source(site)
           end if
         end if
       end select
+
+      ! Check for rejection
+      if (.not. found) then
+        n_reject = n_reject + 1
+        if (n_reject >= EXTSRC_REJECT_THRESHOLD .and. &
+             real(n_sample, 8)/n_reject <= EXTSRC_REJECT_FRACTION) then
```

Comment:
`n_sample` is really the number of *accepted* samples. I'll change the variable name so it's more obvious.